### ISSUE TITLE
tsc does not error mixing type and actual variable in same import statement  on importsNotUsedAsValues=error

### ISSUE THREAD

**Issue by saneyuki** [microsoft/TypeScript#36959]
**TypeScript Version:**  3.8.2

**Search Terms:**

tsc with `importsNotUsedAsValues=error` does not error that mixing type and actual variables in same import statement (e.g. `import { TypeName, ActualVariable } from '.....';`)

**Code**

Set options is `importsNotUsedAsValues=error`

```ts
// bar.ts
export class Bar {};
export function getBar(): Bar {
  return new Bar();
}
```

```ts
// main.ts
import { Bar, getBar } from './bar';

const a: Bar = getBar();
console.log(a);
```

**Expected behavior:**

* In _main.ts_, `import { Bar, getBar } from './bar';` should be error because `Bar` is used only as type. tsc should suggest to split into 2 line like:

```ts
import type { Bar,  } from './bar';
import { getBar } from './bar';
```

**Actual behavior:**

In _main.ts_, tsc does not cause error about using `import type {} from ...`.

If we change _main.ts_ to the following, tsc make it error correctly.

```ts
// main.ts


import { Bar } from './bar'; // By tsc: This import is never used as a value and must use 'import type' because the 'importsNotUsedAsValues' is set to 'error'.ts(1371)`
import { getBar } from './bar';

const a: Bar = getBar();
console.log(a);
```

**Playground Link:** 

I'm sorry. I could not find a way to create this snippet in TypeScript PlayGround.

**Related Issues:**

**Comment by RyanCavanaugh** [comment]
This is the intended behavior. The intent of the flag is to tell you about import *declarations* that would "go missing" during type-only elision (thus potentially changing runtime behavior), not to tell you about *clauses* that would be removed (which have no effect).
